numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1429,19 +1435,28 @@ arr_add_docstring(PyObject *NPY_UNUSED(dummy), PyObject *args)
 
     if (PyGetSetDescr_TypePtr == NULL) {
         /* Get "subdescr" */
-        myobj = PyDict_GetItemString(tp_dict, "fields");
+        myobj = _PyDict_GetItemStringWithError(tp_dict, "fields");
+        if (myobj == NULL && PyErr_Occurred()) {
+            return NULL;
+        }
         if (myobj != NULL) {
             PyGetSetDescr_TypePtr = Py_TYPE(myobj);
         }
     }
     if (PyMemberDescr_TypePtr == NULL) {
-        myobj = PyDict_GetItemString(tp_dict, "alignment");
+        myobj = _PyDict_GetItemStringWithError(tp_dict, "alignment");
+        if (myobj == NULL && PyErr_Occurred()) {
```

Comment:
I suppose I could add `NPY_UNLIKELY` to all these, but that makes things rather verbose. I could also add:
```
#define error_converting_obj(x) NPY_UNLIKELY(x == NULL && PyErr_Occurred()
```
to complement the existing `error_converting(x_numeric)` macro.

Reply:
Don't worry about the `NPY_UNLIKELY` first of all, it optimizes things at the level of 0.2% probably, or less... Second of all. If you care about speed, you should probably intern the string to begin with and not use `GetItemString` at all.

I am good with the helper function/macro if you feel it makes things a bit more readable though.

Reply:
And actually, I think it only optimizes things in any case if there is a bunch of code. But there is only a return or maybe goto there, so it there is no code flow optimization possible to begin with (i.e. much a bunch of code that does not get executed to the back of the function and jump there, so that the normal code flow is contiguous instead of having to jump the bunch of code)

Reply:
I'm happy to leave as is if you are